content/blog/2024/kubewarden-1-10-release.md

```diff
@@ -0,0 +1,135 @@
+for each policy deployed.
+
+Now, memory consumption is constant no matter how many parallel policy
+evaluators are spawned when scaling horizontally, by increasing the number of
+threads available in the machine. This is achieved by only creating an instance
+of the policy evaluator per unique type of policy module.
+
+For example, if a user has the apparmor policy deployed 5 times (with different
+names, settings, etc), now only one instance of PolicyEvaluator will be
+allocated for all of those 5 deployments.
+
+In addition, we rely now on Wasmtime's pre-initialization of Wasm modules. This
+allows us to keep a deduplicated list of all the policies in use, in their
+pre-initialized form. This list is shared between all policy evaluators in the
+policy-server.
+
+#### Protection against policy memory leaks, and clean slate per policy
+
+Before, there were long-running policy evaluators, ready to process policy requests.
+
+Now, policy evaluation mode happens on-demand: we receive a request, a fresh
+policy evaluator is created, used just for evaluation that request, and
+discarded.
+
+This is possible as we now leverage Wasmtime's pre-initialization, hence
+creation time of policy evaluators in greatly reduced.
+
+Using this on-demand approach to policy evaluators brings two main advantages:
+
+- Impact of policies leaking memory is now virtualy non-existent.
+- Each evaluation starts with a clean slate, preventing bugs caused by policies
+  leaving unclean state between evaluations.
+
+#### Codebase simplification and increased test coverage
+
+Also, thanks to a big refactor, together with additional unit and integration
+tests, the policy-server codebase has moved away from using a worker thread
+pool for policy evaluation to an asynchronous approach, removing codebase complexity.
+
+Stay tuned for more performance improvements!
+
+## WASI policies written in Go can now be context aware
+
+Due to recent improvements in the official Go compiler, we have added
+context-aware capabilities, via WASI, to the Kubewarden Go SDK, so it is
+now possible to write context-aware Go WASI policies.
+
+No divergence is needed when writing context-aware policies in WASI. Just use the Go SDK as usual,
+making use of its `capabilities` package (which exposes quering for kubernetes
+resources, OCI manifest digest, Sigstore signature verification, etc).
+
+As an example, have a look at the
+[go-wasi-context-aware-test-policy](https://github.com/kubewarden/go-wasi-context-aware-test-policy).
+
+## Controller changes
+
+The controller now reconciles divergences between `ClusterAdmissionPolicies`
+and `AdmissionPolicies` and their associated webhook objects. This helps mitigate
+against attacks that delete or modify policy webhooks, preventing policies from working correctly. RBAC rights already mitigate against these attacks.
+
+## Audit Scanner changes
+
+Thanks to the community contributions from Nuno Nelas and Bruno René Santos, the
```

Comment:
@nnelas @brunorene are you fine mentioning you by name here?

Reply:
hallo @jvanz ! I'm also okay with it, thank you :D